src/transport/AdminPairingTable.cpp

```diff
@@ -28,6 +28,18 @@ namespace chip {
 
 namespace Transport {
 
+CHIP_ERROR AdminPairingInfo::SetFabricLabel(uint8_t * fabricLabel)
+{
+    char *fabricLabelChar = Uint8::to_char(fabricLabel);
+    if (strlen(fabricLabelChar) + 1 > 32)
```

Comment:
Need to use strnlen to detect lack of nul-termination without moving past end of array. Otherwise this is a possibility of buffer overrun.